scripts/release/asset_publish.py

```diff
@@ -563,13 +584,13 @@ def _str2bool(v: str) -> bool:
                     component_type = asset.spec_as_object().type
                     if component_type == assets.ComponentType.PIPELINE.value:
                         if not validate_and_prepare_pipeline_component(
-                            asset.spec_with_path, passed_version, registry_name
+                            asset.spec_with_path, passed_version, registry_name, validate_assets
```

Comment:
Why pass it into the function only to have it exit immediately? Just use it here to skip the call to `validate_and_prepare_pipeline_component` instead.

Reply:
Crossed my mind as well but thought too many if conditions in that part of code. Updated now. Let me know if you have any suggestions.